packages/expo-modules-autolinking/src/autolinking/findModules.ts

```diff
@@ -41,6 +43,23 @@ export async function findModulesAsync(providedOptions: SearchOptions): Promise<
         fallbackToDirName: isNativeModulesDir,
       });
 
+      // Check if the project is using isolated modules, by checking
+      // if the parent dir of `packagePath` is a `node_modules` folder.
+      // Isolated modules installs dependencies in small groups such as:
+      //   - /.pnpm/expo@50.x.x(...)/node_modules/@expo/cli
+      //   - /.pnpm/expo@50.x.x(...)/node_modules/expo
+      //   - /.pnpm/expo@50.x.x(...)/node_modules/expo-application
+      // When isolated modules are detected, expand the `searchPaths`
+      // to include possible nested dependencies.
+      const maybeIsolatedModulesPath = path.join(
+        packagePath,
+        name.startsWith('@') && name.includes('/') ? '../..' : '..' // scoped packages are nested deeper
+      );
+      const isIsolatedModulesPath = path.basename(maybeIsolatedModulesPath) === 'node_modules';
+      if (isIsolatedModulesPath && !searchPaths.has(maybeIsolatedModulesPath)) {
```

Comment:
a dumb question: if a package is not hoisted by yarn or npm package managers, does it fall into this `isIsolatedModulesPath` check?
e.g. `node_modules/expo/node_modules/expo-application`.

Reply:
Definitely not a dumb question, and yes, code will be triggered to visit the modules within `node_modules/expo/node_modules` by adding this path to the `searchPaths` set.

In some ways this makes sense, since it's not detected by autolinking _without_ this fix. (e.g. `duplicates` is not populated with the version from `node_modules/expo/node_modules/expo-application`). With these duplicate results, we can inform users better about possible version mismatches in the project.

Eventually, since this is a nested dependency, it does NOT take precedence over `node_modules/expo-application` - this one is used instead and tested in the added test cases. But, Metro might still use the JS code of `node_modules/expo/node_modules/expo-application`.

Reply:
the code doesn't have any problems. i'm just double checking my understanding. thanks for the great answer.

Reply:
Was just thinking out loud, so wasn't sure if it made sense either 😅 But I do think so.